compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/XCLBinGen.cpp

```diff
@@ -1375,69 +1376,116 @@ LogicalResult generateUnifiedObject(
   return success();
 }
 
-/// Assume the ELF files have already been generated and are stored in the
-/// `tempDirPath`. This function converts `xilinx::aie::device` to
-/// `amdaie.npu.control_packets` by calling the
-/// `AMDAIEConvertDeviceToControlPacketsPass`.
-LogicalResult generateControlPackets(MLIRContext *context,
-                                     AIE::DeviceOp deviceOp,
-                                     const Path &tempDirPath,
-                                     bool printIRBeforeAll,
-                                     bool printIRAfterAll,
-                                     bool printIRModuleScope, bool timing) {
+}  // namespace
+
+namespace mlir::iree_compiler::AMDAIE {
+
+/// Pipeline to generate control packets from `xilinx::aie::device`, and dump
+/// them into files.
+LogicalResult generateControlPackets(
+    MLIRContext *context, AIE::DeviceOp deviceOp,
+    const IREE::HAL::ExecutableTargetAttr &targetAttr, const Path &tempDirPath,
```

Comment:
```suggestion
    IREE::HAL::ExecutableTargetAttr targetAttr, const Path &tempDirPath,
```
Attributes are typically passed by value, they are small wrappers around a pointer, so the same size.

Also, can't you just get the target attribute from the DeviceOp?